.gitignore

```diff
@@ -42,4 +42,4 @@
 !/var/package
 /var/package/*
 !/var/package/*.xml
-
+composer.lock
```

Comment:
You need the lock file to ensure that people is working always with the same dependencies.

Rather than ignoring the .lock file in the repository, can't you ignore it in the package? Where is the warning coming from?

Reply:
We cannot track the changes in dependencies that are listed in composer.lock file. For example, in our case, laminas/laminas-http is updated with the new version which is a dependency of the magento/framework and it has its own lock file.

So we should not add a lock file that affects other dependencies.

Requiring or updating the composer will fetch the latest dependencies automatically so in our case lock file is not required.